specification/automation/resource-manager/Microsoft.Automation/preview/2022-06-30-preview/runbook.json

```diff
@@ -0,0 +1,1693 @@
+          "description": "Gets or sets the provisioning state of the runbook.",
+          "enum": [
+            "Succeeded"
+          ],
+          "x-ms-enum": {
+            "name": "RunbookProvisioningState",
+            "modelAsString": false
+          }
+        },
+        "lastModifiedBy": {
+          "type": "string",
+          "description": "Gets or sets the last modified by."
+        },
+        "creationTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Gets or sets the creation time.",
+          "x-nullable": false
+        },
+        "lastModifiedTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Gets or sets the last modified time.",
+          "x-nullable": false
+        },
+        "description": {
+          "type": "string",
+          "description": "Gets or sets the description."
+        }
+      },
+      "description": "Definition of the runbook property type."
+    },
+    "Runbook": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/RunbookProperties",
+          "x-ms-client-flatten": true,
+          "description": "Gets or sets the runbook properties."
+        },
+        "etag": {
+          "type": "string",
+          "description": "Gets or sets the etag of the resource."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../common/v1/definitions.json#/definitions/TrackedResource"
+        }
+      ],
+      "description": "Definition of the runbook type."
+    },
+    "RunbookListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Runbook"
+          },
+          "description": "Gets or sets a list of runbooks."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Gets or sets the next link."
+        }
+      },
+      "description": "The response model for the list runbook operation."
+    },
+    "RunbookCreateOrUpdateProperties": {
+      "type": "object",
+      "properties": {
+        "logVerbose": {
+          "type": "boolean",
+          "description": "Gets or sets verbose log option."
+        },
+        "logProgress": {
+          "type": "boolean",
+          "description": "Gets or sets progress log option."
+        },
+        "runbookType": {
+          "type": "string",
+          "description": "Gets or sets the type of the runbook.",
+          "enum": [
+            "Script",
+            "Graph",
+            "PowerShellWorkflow",
+            "PowerShell",
+            "GraphPowerShellWorkflow",
+            "GraphPowerShell",
+            "Python2",
+            "Python3",
+            "Python"
+          ],
+          "x-ms-enum": {
+            "name": "RunbookTypeEnum",
+            "modelAsString": true
+          }
+        },
+        "runtime": {
```

Comment:
>runtime [](http://example.com/codeflow?start=9&length=7)

please rename to runtiumeResourceId so it is clear that this is an arm resource id

Reply:
We had a similar comment in another PR for the same resource which was later abandoned. https://github.com/Azure/azure-rest-api-specs/pull/19662#discussion_r913887638

The property is just name and is the new runtime resource and it would be validated in RP if the name is correct.
The runtime resource would not be used outside the automation account context. So, fully qualified resource would rather be an overhead for customer to add in PUT/PATCH calls. Also, RP ensures uniqueness of runtime name for an automation account.

Just the resource name suits more to the scenario.